### ISSUE TITLE
Problems with Gatsby localised 404 and Now routing

### ISSUE THREAD

**Issue by michaellopez** [vercel/vercel#3207]
Hi Zeit,

Cool to see you launched a Gatsby plugin, kudos.

We're in the middle of working out a multilingual site on Gatsby and we're currently hitting a problem with handling 404 pages on Now.

We generate our 404 pages based on the locales defined in our headless CMS. So Gatsby sources from our headless CMS, dynamically generates a 404 page **per locale** with locale specific content and outputs each respective page to `/<locale>/404`, for example: `/en/404`, `/de/404`, `/sv/404`, etc.

We've tried all our Now-routing kung fu that we could think up to handle these paths dynamically and properly on Now, but we can't seem to get it right. Our definition of "properly" is:

- All 404 pages, regardless if it is a locale 404 or default Gatsby 404 should have HTTP status code of 404
- Zeit Now's 404 page should never display, our default 404 page from Gatsby should display if there isn't any locale specific 404 page

So here is a small sample of what we want when locales `en` and `de` are defined in our headless CMS and `sv` is **NOT** defined:
```
/en/page-exist => HTTP 200 Gatsby page
/en/does-not-exist => HTTP 404 Gatsby **EN Locale** 404
/de/does-not-exist => HTTP 404 Gatsby **DE Locale** 404
/sv => HTTP 404 Gatsby 404 page (no locale)
/sv/does-not-exist => HTTP 404 Gatsby 404 page (no locale)
/does-not-exist => HTTP 404 Gatsby 404 page (no locale)
/api/does-not-exist => HTTP 404 Gatsby 404 page (no locale)
/api/lamdba-does-exist => HTTP 200 Lamdba result (or whatever status code the lambda returns)
```
As I said, we've tried many different configs but were unable to get all of the above to work. Either the HTTP status code was not 404, or Zeit Nows 404 page was displayed. I'll spare you the details but it seems that `"handle": "filesystem"` short-circuits on `"status": 404` and just returns Zeit Nows 404 page, instead of checking the filesystem for the `dest` page and returning that with `"status": 404` when having a route with `"continue": true` before it.

The only way we see getting it to work is to define a route entry in `now.json` for each available locale defined in our headless CMS. This does not work that well for us though, because our content editors would require a developer to update `now.json` whenever a new locale is introduced in the headless CMS and launched, making for a not so streamlined workflow.

After seeing the introduction of `__now_routes_g4t5bY.json` in `now-static-build` I got excited because that means we can now generated our locale routes dynamically based on available data in Gatsby (that is sourced from our headless CMS) and have `now-static-build` pick them up during deploy.

So now for my question: Are you interested in a PR to `gatsby-plugin-now` to add handling for this? Or should we do this in our own gatsby plugin? Or are you interested in the details of our different routing config approaches to help improve/add handling to Now routing to handle these issues for frameworks other than Gatsby? I don't know if other frameworks/people deploying to Now are experiencing our problems, because Google can't find them at least. We can't be the only once doing multilingual sites and concerned with proper handling of 404 and HTTP status codes.

Thank you for your time! 🍰

**Comment by michaellopez** [comment]
We have reproduction repo without the dynamic creation of 404 pages from our headless CMS. It showcases the issue.

Repo: https://gitlab.com/lofgrenfredrik/routing-test
Deploy: https://routing-test.lofgrenfredrik.now.sh

For example
- https://routing-test.lofgrenfredrik.now.sh/en/does-not-exist should not display Zeit Now 404 - instead it should display Gatsbys default 404 page without localisation.
- https://routing-test.lofgrenfredrik.now.sh/api/does-not-exist should not display Zeit Now 404 - instead it should display Gatsbys default 404 page without localisation. 
- https://routing-test.lofgrenfredrik.now.sh/gb/does-not-exist - works like intended, http 404 and  Gatsby localised 404 page for GB locale
- https://routing-test.lofgrenfredrik.now.sh/api/lambda - works like indended, http 200 from lambda

**Comment by michaellopez** [comment]
The 404 for the lambdas comes from https://github.com/zeit/now/blob/49bd2439a7d7f835924f911f8ecff40b1c1b2918/packages/now-build-utils/src/detect-routes.ts#L260,L266

Could that be made optional and/or configurable?

**Comment by michaellopez** [comment]
It just hit me, maybe we aren't the zero config use case 😅

I don't see how the ability to disable the Zeit Now 404 page for non existent lambdas should be handled in a zero config environment, as it is config that is needed to disable it 😆 any ideas?

We'll take a stab at solving this without zero config and see where we end up...

**Comment by michaellopez** [comment]
Currently we are doing some testing without "Zero Config" to see if we can manage to make it work that way. Because our use case might not be considered a "Zero Config" use case. What do you think?

**Comment by leerob** [comment]
Hey! This issue has gone stale. If you're still having issues here, could you please open a new issue? 

Thank you so much 🙏
